src/components/scenes/WcConnectionsScene.tsx

```diff
@@ -40,6 +41,7 @@ export const WcConnectionsScene = (props: Props) => {
   const styles = getStyles(theme)
   const [connections, setConnections] = React.useState<WcConnectionInfo[]>([])
   const [connecting, setConnecting] = React.useState(false)
+  const [sessionProposal, setSessionProposal] = React.useState<{ [uri: string]: Web3WalletTypes.SessionProposal }>({})
```

Comment:
Please use a map for this! Since the URL is attacker-controlled input, we absolutely cannot use it as an object key.